shell/platform/darwin/graphics/FlutterDarwinExternalTextureMetal.mm

```diff
@@ -116,29 +139,36 @@ - (void)onTextureUnregistered {
 
 #pragma mark - External texture skia wrapper methods.
 
-- (sk_sp<SkImage>)wrapExternalPixelBuffer:(CVPixelBufferRef)pixelBuffer
-                                grContext:(GrDirectContext*)grContext {
+- (sk_sp<flutter::DlImage>)wrapExternalPixelBuffer:(CVPixelBufferRef)pixelBuffer
+                                           context:(flutter::Texture::PaintContext&)context {
   if (!pixelBuffer) {
     return nullptr;
   }
 
-  sk_sp<SkImage> image = nullptr;
+  sk_sp<flutter::DlImage> image = nullptr;
   if (_pixelFormat == kCVPixelFormatType_420YpCbCr8BiPlanarVideoRange ||
       _pixelFormat == kCVPixelFormatType_420YpCbCr8BiPlanarFullRange) {
-    image = [self wrapNV12ExternalPixelBuffer:pixelBuffer grContext:grContext];
+    image = [self wrapNV12ExternalPixelBuffer:pixelBuffer context:context];
   } else {
-    image = [self wrapRGBAExternalPixelBuffer:pixelBuffer grContext:grContext];
+    image = [self wrapRGBAExternalPixelBuffer:pixelBuffer context:context];
   }
 
   if (!image) {
-    FML_DLOG(ERROR) << "Could not wrap Metal texture as a Skia image.";
+    FML_DLOG(ERROR) << "Could not wrap Metal texture as a display list image.";
   }
 
   return image;
 }
 
-- (sk_sp<SkImage>)wrapNV12ExternalPixelBuffer:(CVPixelBufferRef)pixelBuffer
-                                    grContext:(GrDirectContext*)grContext {
+- (sk_sp<flutter::DlImage>)wrapNV12ExternalPixelBuffer:(CVPixelBufferRef)pixelBuffer
+                                               context:(flutter::Texture::PaintContext&)context {
+#if IMPELLER_SUPPORTS_RENDERING
+  if (_enableImpeller) {
+    IMPELLER_UNIMPLEMENTED
```

Comment:
Can we add a validation log here instead and also add a followup bug for its implementation?

Reply:
Done.